File: frontend/src/services/markService.js
import api from './api'

export const markService = {
  create: (payload) => api.post('/marks', payload),
  findByStudent: (studentId) => api.get(`/marks/student/${studentId}`),
  update: (id, payload) => api.put(`/marks/${id}`, payload),
  remove: (id) => api.delete(`/marks/${id}`),
}

File: backend/src/main/java/com/school/service/MarkService.java
package com.school.service;

import com.school.dto.request.MarkRequest;
import com.school.dto.response.MarkResponse;

import java.util.List;

public interface MarkService {
    MarkResponse create(MarkRequest request, String enteredByTeacherId);
    List<MarkResponse> findByStudent(String studentId);
    MarkResponse update(String id, MarkRequest request);
    void delete(String id);
}